### PR TITLE
feat: Public dashboard security rules

### LINKED ISSUE #1557: [Public Dashboard] Update Firestore security rules
**Description**
As we want to add a new user authentication state (anonymous user), we should update existing Firestore security rules.

**Acceptance Criteria**
- [ ] Add new Firestore security rules according to the design.
- [ ] Test the rules.

**Additional context**
[Public dashboard design](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/features/public_dashboard/01_public_dashboard.md#design)

### PR DESCRIPTION
**Pull request by solid-maxim**
Fixes #1557 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/firebase/firestore/rules/firestore.rules [hunk 1]
```diff
@@ -68,11 +69,21 @@ service cloud.firestore {
...
+
   /// Checks if the sign in provider from the request is a password.
   function isSignInProviderPassword() {
-    let authToken = request.auth.token;
+    return checkSignInProvider('password');
+  }
 
-    return authToken.firebase.sign_in_provider == 'password';
+  /// Checks if the sign in provider from the request is an anonymous.
+  function isSignInProviderAnonymous() {
```
Comment (illia-romanenko) [inline metrics/firebase/firestore/rules/firestore.rules:85]: Should we check that the public dashboard is enabled too?

File: metrics/firebase/firestore/rules/firestore.rules [hunk 2]
```diff
@@ -3,30 +3,31 @@ service cloud.firestore {
...
     match /build/{buildId} {
-      allow read: if isAccessAuthorized(database);
+      allow read: if isAccessAuthorized(database) || isSignInProviderAnonymous();
       allow create, update: if isAccessAuthorized(database) && isBuildValid(database);
       allow delete: if false;
     }
 
     match /build_days/{buildDayId} {
-      allow read: if isAccessAuthorized(database);
+      allow read: if isAccessAuthorized(database) || isSignInProviderAnonymous();
```
Comment (illia-romanenko) [inline metrics/firebase/firestore/rules/firestore.rules:24]: Should we improve naming a bit to make sure that it's readable & understandable how those 2 checks are different and what are the use cases?

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Looks good. Thanks, @solid-maxim. Added a couple of comments - let me know what you think. Glad that the structure that we have allows adding such checks so easily.

**Comment by solid-anatolyeva** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request closed without merging**